### DISCUSSION TITLE
Adaptations to the Taskwarrior Tools page

### DISCUSSION THREAD

**Discussion by lauft** [GothenburgBitFactory/taskwarrior#3490] [Announcements]
With the release of Taskwarrior 3 and its changes to task synchronization some changes to the crawler script became necessary.

Currently, the content of the [Taskwarrior Tools](https://taskwarrior.org/tools) page is generated by querying the GitHub API for repositories containing the keyword `taskwarrior` or `taskserver` in their name, description, or as topic.

As Taskwarrior 3 has a different way of synchronizing, some tools are no longer compatible with it. To better indicate this on the [Taskwarrior Tools](https://taskwarrior.org/tools) page, we want to assign categories to the tools and provide means to filter for them. 

To implement this, the GitHub API query will no longer search names or descriptions, but only topics for the keywords. The categories are then set according to the found topics (this also helps us to reduce the number of overall queries to the GitHub API, improving speed and performance).

This requires some adaptations of the listed repositories as well. However, we want to make sure that repositories which do not apply them are not dropped from the list. It may happen though that their listing information will no longer be updated and stay frozen.

If your tool is intended for
* Taskwarrior 3, add topic `taskwarrior3` to your repository
* Taskwarrior 2, add topic `taskwarrior2` to your repository
* Taskserver, add topic `taskserver` to your repository

Those topics are not exclusive. The already present topic `taskwarrior` will then mean your tool is intended for both, Taskwarrior 2 and Taskwarrior 3, so you can also use this (or keep it). Tools without any of these topics set will be categorized as `taskwarrior2` and `taskserver`.

**Comment by lauft** [comment]
The tool category groups (a.k.a. "intentions") are now displayed as icons on the [tools page](https://taskwarrior.org/tools). Filtering will come at a later stage.

**Comment by xingheng** [comment]
What if adding both `taskwarrior` and `taskwarrior2` topics at the same time...

**Reply by lauft** [reply]
In that case, the topic `taskwarrior` would "win" and your tool would be categorized as "intended for both, Taskwarrior 2 and Taskwarrior 3"
